src/browser/history.js

```diff
@@ -0,0 +1,321 @@
+  };
+
+
+/**
+
+### Quotes
+
+Quotes store contains records of quotes that have being created by a user that
+have a following structure.
+
+
+{
+  _id: "quote/W29iamVjdCBBcnJheUJ1ZmZlcl0=",
+  content: `If you say that <span class="fixed">a</span> is 5, you can't say it's something else later because you just said it was 5. What are you, some kind of liar? So in purely functional languages, a function has no side-effects. The only thing a function can do is calculate something and return it as a result.`
+  uri: "http://learnyouahaskell.com/introduction#about-this-tutorial",
+  tags: ["functional"]
+}
+**/
+
+  const Quote = Record({
+    [PouchDB.id]: ID,
+    [PouchDB.revision]: Maybe(Revision),
+
+    type: Type('Quote'),
+    uri: URI,
+    content: String,
+    tags: List(TagName)
+  });
+  Quote.storeID = 'quotes';
+
+  Quote.construct = async(function* ({uri, content}) {
+    const hash = yield sha(content);
+    return Quote({[PouchDB.id]: `quote/${uri}/${hash}`, uri, content});
+  });
+
+
+  /**
+  {
+    _id: "tag/haskell",
+    description: "Haskell programming language",
+    name: "haskell",
+    items: [
+      "quote/W29iamVjdCBBcnJheUJ1ZmZlcl0="
+      "page/http://learnyouahaskell.com"
+    ]
+  }
+  **/
+
+  const Tag = Record({
+    [PouchDB.id]: ID,
+    [PouchDB._rev]: Maybe(Revision),
+    type: Type("Tag"),
+    name: TagName,
+    items: List(ID)
+  });
+  Tag.storeID = 'tags';
+
+  Tag.add = (tag, item) =>
+    tag.update('items', include(item[PouchDB.id]));
+
+  Tag.remove = (tag, item) =>
+    tag.update('items', exclude(item[PouchDB.id]));
+
+  Tag.tag = (item, tagName) => item.update('tags', include(tagName));
+
+  Tag.untag = (item, tagName) => item.update('tags', exclude(tagName));
+
+  const Top = Record({
+    [PouchDB.id]: ID('top/pages'),
+    [PouchDB.revision]: Maybe(Revision),
+    pages: List(Page)
+  });
+  Top.storeID = Page.storeID;
+  Top.sample = (page, limit) => top => top.update('pages', pages => {
+    const index = pages.findIndex(x => x[PouchDB.id] === page[PouchDB.id])
+    return pages.set(index < 0 ? pages.size : index, page)
+                .sortBy(Page.frequency)
+                .take(limit);
+  });
+
+  const clear = async(function*({stores}) {
+    yield stores.pages.destroy();
+    yield stores.quotes.destroy();
+    yield stores.tags.destroy();
+  });
+
+  let taskID = 0;
+
+  const scheduleEdit = async(function*(history, record, transform) {
+    try {
+      // wait for last scehduled edit to complete.
+      yield history.editQueue[record[PouchDB.id]];
+    } catch (error) {
+      // If last operation faild log an error, but proceed to a next
+      // operation.
+      console.error(error);
+    } finally {
+      const store = history.stores[record.constructor.storeID];
+      return edit(record, transform, store);
+    }
+  })
+
+  // History
+  class History {
+    static defaults() {
+      return {
+        pagesStoreName: "pages",
```

Comment:
Oh yeah I wrote most of this code before we decided to use single quotes over doubles.